src/libraries/System.Linq/src/System/Linq/Repeat.SpeedOpt.cs

```diff
@@ -28,10 +28,7 @@ public TResult[] ToArray()
             public List<TResult> ToList()
             {
                 List<TResult> list = new List<TResult>(_count);
-                for (int i = 0; i != _count; ++i)
-                {
-                    list.Add(_current);
-                }
+                SetCountAndGetSpan(list, _count).Fill(_current);
```

Comment:
> It would be making assumptions on how the CollectionsMarshal.SetCount implementation surfaces uninitialized elements however.

Right, my concern was if List was changed to use eg GC.AllocateUninitialozedArray that would silently end up making this wrong.